File: backend/api/app/models/preference_profile.rb
# frozen_string_literal: true

class PreferenceProfile < ApplicationRecord
  belongs_to :user

  enum temperature_preference: { cool: 0, warm: 1, no_preference: 2 }, _prefix: true
  enum movement_preference: { very_little: 0, moderate: 1, a_lot: 2, as_much_as_possible: 3 }, _prefix: true
  enum self_chronotype: { morning: 0, night: 1, depends: 2 }, _prefix: true

  validates :sleep_together_importance, inclusion: { in: 1..5 }, allow_nil: true
  validates :rhythm_importance, inclusion: { in: 1..5 }, allow_nil: true
end

File: apps/ios/Synca/Models/PreferenceProfile.swift
import Foundation

enum TemperaturePreference: String, Codable, CaseIterable {
    case cool, warm
    case noPreference = "no_preference"
}

enum MovementPreference: String, Codable, CaseIterable {
    case veryLittle = "very_little"
    case moderate
    case aLot = "a_lot"
    case asMuchAsPossible = "as_much_as_possible"
}

enum SelfChronotype: String, Codable, CaseIterable {
    case morning, night, depends
}

/// Backs the 5-question declared-preferences questionnaire, which — per the design
/// review — also doubles as the Spark on-the-spot questionnaire. One field per
/// question, in the order defined in docs/features/signals-v1.md § Questionnaire.
struct PreferenceProfile: Codable, Equatable {
    var sleepTogetherImportance: Int?
    var temperaturePreference: TemperaturePreference?
    var movementPreference: MovementPreference?
    var rhythmImportance: Int?
    var selfChronotype: SelfChronotype?
}

/// `PATCH /signals/preferences` request body wraps under `preferences:`.
struct PreferenceProfileUpdateRequest: Codable {
    let preferences: PreferenceProfile
}
